kratos/future/input_output/vtu_output.cpp

```diff
@@ -0,0 +1,1689 @@
+    Globals::DataLocation DataLocation)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(mIsPVDFileHeaderWritten)
+        << "Flags can be added only before the first call to the PrintOutput [ flag name = "
+        << rFlagName << " ].\n";
+
+    switch (DataLocation) {
+        case Globals::DataLocation::NodeNonHistorical:
+            // user is trying to add a flag to nodes. now we are checking
+            // whether there are variables with the same name in the variables
+            // from the nodal historical.
+            // note: there is no break here.
+            CheckDataArrayName(rFlagName, {Globals::DataLocation::NodeHistorical}, mFlags, mVariables);
+        case Globals::DataLocation::Condition:
+        case Globals::DataLocation::Element:
+            // user is trying to add a flag variable to nodes, conditions or elements.
+            // here we check if the given flag name is there in the existing flags for user specified data location,
+            // and also in the variables.
+            CheckDataArrayName(rFlagName, {DataLocation}, mFlags, mVariables);
+
+            // since specified flags are used to output in every unstructured grid. We check here whether
+            // any of the tensor adaptors in the given data location contains the same name as rFlagName.
+            CheckDataArrayName(rFlagName, DataLocation, mUnstructuredGridDataList);
+
+            // If there are no conflicts, we add the flag variable.
+            mFlags[static_cast<IndexType>(DataLocation)][rFlagName] = &rFlagVariable;
+            break;
+        default:
+            KRATOS_ERROR << "Flags can be only added to NodeNonHistorical, Condition, and Element data locations.";
+            break;
+    }
+
+    KRATOS_CATCH("");
+}
+
+void VtuOutput::AddVariable(
+    SupportedVariablePointerType pVariable,
+    Globals::DataLocation DataLocation)
+{
+    KRATOS_TRY
+
+    KRATOS_ERROR_IF(mIsPVDFileHeaderWritten)
+        << "Variables can be added only before the first call to the PrintOutput [ variable name = "
+        << GetName(pVariable) << " ].\n";
+
+    switch (DataLocation) {
+        case Globals::DataLocation::NodeHistorical:
+            // Now the user is adding a nodal historical variable. So, this checks whether
+            // the a variable with the same name exists in the nodal non-historical variables map.
+            // note: there is no break in here.
+            CheckDataArrayName(GetName(pVariable), {Globals::DataLocation::NodeNonHistorical}, mFlags, mVariables);
+        case Globals::DataLocation::NodeNonHistorical:
+            // Now the user is adding a nodal non-historical variable. So this checks whether
+            // a variable with the same name exists in the nodal historical variables map.
+            // not: there is no break in here.
+            CheckDataArrayName(GetName(pVariable), {Globals::DataLocation::NodeHistorical}, mFlags, mVariables);
+        case Globals::DataLocation::Condition:
```

Comment:
it is c++ but it is really counterintutitive that if you skip a case you go and execute the following ... can you add a comment?